include/dlaf/eigensolver/eigensolver.h

```diff
@@ -19,6 +19,29 @@
 
 namespace dlaf::eigensolver {
 
+/// Standard Eigensolver.
+///
+/// It solves the standard eigenvalue problem A * x = lambda * x.
+///
+/// On exit, the lower triangle or the upper triangle (depending on @p uplo) of @p mat_a,
+/// including the diagonal, is destroyed.
+///
+/// Implementation on local memory.
+///
+/// @param uplo specifies if upper or lower triangular part of @p mat will be referenced
+/// @param mat contains the Hermitian matrix A
+/// @param eigenvectors matrix of eigenvectors
+/// @param eigenvalues vector of eigenvalues
+template <Backend B, Device D, class T>
+void eigensolver(blas::Uplo uplo, Matrix<T, D>& mat, Matrix<T, D>& eigenvectors,
+                 Matrix<BaseType<T>, D>& eigenvalues) {
```

Comment:
Is this more consistent with the other overload? I'm not quite sure myself if this is actually better...
```suggestion
void eigensolver(blas::Uplo uplo, Matrix<T, D>& mat, EigensolverResult<T, D>& result) {
```
If anything the `eigenvalues` and `eigenvectors` should possibly be passed in the same order as they're declared in `EigensolverResult`?